cptoml.py

```diff
@@ -156,57 +167,51 @@ def _tablefind(buf, subtable) -> int:
 
     Returns -1 when not found.
     """
-    result = -1
-    tm = 0
-    q = True
-    while q:
-        tml = None
+
+    for index, line in enumerate(buf):
         try:
-            tml = _prepareline(buf[tm])
+            tml = _prepareline(line)
+
+            if tml == f"[{subtable}]":
+                return index
+
         except IndexError:
-            q = False
-        if q and tml.startswith("[") and (tml == f"[{subtable}]"):
-            result = tm
-            q = False
-        del tml
-        tm += 1
-        if not q:
             break
-    del buf, subtable, tm, q
-    return result
+
+    del buf, subtable
+    return -1
 
 
 def _getkeys(buf, start=0):
     """
     Get the keys off a table.
     """
-    res = list()
-    q = True
+    res = []
     tm = start
-    while q:
+    for index, line in enumerate(buf, start):
```

Comment:
This could do more loops than needed?
And perhaps do dupe data in case of the key being there twice?

Reply:
There's no such thing as "more than needed" when you want to get all keys in the table, right?

As per the duped keys, perhaps we should store the keys in a `set` instead of a list, to ensure unique-ness, could also fit a check to warn/error if duplication is found